# Medidas de Tendencia Central y Medidas de Variación
Ejercicios de medidas de tendencia central y variación

# 

In [62]:
import pandas as pd
from statistics import mean, median, mode, variance, stdev

##### Tiempos de Espera de Clientes
A continuación se presentan los tiempos de espera (en minutos) de los clientes del banco Jefferson Valley (donde todos los clientes forman una sola fila) y del banco Providence (donde los clientes esperan en filas individuales, en tres ventanillas diferentes). 
+ Determine si existe una diferencia entre los dos conjuntos de datos, que no sea aparente cuando se comparan las medidas de tendencia central. 
+ Si tal diferencia existe, ¿cuál es?
    + Jefferson Valley (una sola fila): 6.5 6.6 6.7 6.8 7.1 7.3 7.4 7.7 7.7 7.7
    + Providence (filas individuales): 4.2 5.4 5.8 6.2 6.7 7.7 7.7 8.5 9.3 10.0

In [63]:
# Muestras
Jeff = [6.5, 6.6, 6.7, 6.8, 7.1, 7.3, 7.4, 7.7, 7.7, 7.7]
Prov = [4.2, 5.4, 5.8, 6.2, 6.7, 7.7, 7.7, 8.5, 9.3, 10.0]

In [64]:
# Medidas de tendencia central. caldulo de media, mediana y moda
print(f"a. Jefferson  Central Tendency Measures: 1) media: {mean(Jeff)},    2) mediana: {median(Jeff)}, 3) mode: {mode(Jeff)}")
print(f"b. Providence Central Tendency Measures: 1) media: {mean(Prov)},    2) mediana: {median(Prov)},                 3) mode: {mode(Prov)}")

a. Jefferson  Central Tendency Measures: 1) media: 7.15,    2) mediana: 7.199999999999999, 3) mode: 7.7
b. Providence Central Tendency Measures: 1) media: 7.15,    2) mediana: 7.2,                 3) mode: 7.7


In [65]:
# Medidas de Variación. Cálculo de Varianza y Desfiación Estandard
print(f"a. Jefferson  1) Varianza: {variance(Jeff)}, 2) Desviación Estandard: {stdev(Jeff)}")
print(f"b. Providence 1) Varianza: {variance(Prov)},  2) Desviación Estandard: {stdev(Prov)}")

a. Jefferson  1) Varianza: 0.22722222222222235, 2) Desviación Estandard: 0.4766783215358365
b. Providence 1) Varianza: 3.3183333333333334,  2) Desviación Estandard: 1.8216293073326784


Aunque las medidas de tendencia central son las mismas, los tiempos de espera de Providence son mucho más variados que los tiempos de espera de Jefferson Valley.
+ Se puede ver en la igualdad de las medidas de tendencia central y en las diferencias de las medidas de variación

________________________________________________________________________________________________________________________________________________________
##### Coca-Cola regular vs Coca-Cola dietética. 
Los siguientes son los pesos (en libras) de muestras del contenido de latas de Coca-Cola regular y dietética. 
+ ¿Parece existir una diferencia significativa entre los dos conjuntos de datos? 
+ ¿Cómo se podría explicar una diferencia como ésta?
    + Regular: 0.8192 0.8150 0.8163 0.8211 0.8181 0.8247
    + Dietética: 0.7773 0.7758 0.7896 0.7868 0.7844 0.7861


In [66]:
# Muestras
Reg = [0.8192, 0.8150, 0.8163, 0.8211, 0.8181, 0.8247]
Diet = [0.7773, 0.7758,  0.7896,  0.7868, 0.7844, 0.7861]

In [67]:
# Medidas de tendencia central
print(f"a. Regular Central Tendency Measures: 1) media: {mean(Reg)},    2) mediana: {median(Reg)},  3) mode: {mode(Reg)}")
print(f"b. Dietética Central Tendency Measures: 1) media: {mean(Diet)},     2) mediana: {median(Diet)},             3) mode: {mode(Diet)}")

a. Regular Central Tendency Measures: 1) media: 0.8190666666666667,    2) mediana: 0.8186500000000001,  3) mode: 0.8192
b. Dietética Central Tendency Measures: 1) media: 0.7833333333333333,     2) mediana: 0.78525,             3) mode: 0.7773


In [68]:
# Diferencia entre ambas muestras
print(f"Regular-Dietética = {[Reg[i] - Diet[i] for i in range(len(Reg))]}.")

Regular-Dietética = [0.04190000000000005, 0.0391999999999999, 0.026700000000000057, 0.0343, 0.03370000000000006, 0.03859999999999997].


In [69]:
# Medidas de Variación. Cálculo de Varianza y Desfiación Estandard
print(f"a. Regular  1) Varianza: {variance(Reg)}, 2) Desviación Estandard: {stdev(Reg)}")
print(f"b. Dietética 1) Varianza: {variance(Diet)},  2) Desviación Estandard: {stdev(Diet)}")

a. Regular  1) Varianza: 1.2202666666666725e-05, 2) Desviación Estandard: 0.0034932315506800756
b. Dietética 1) Varianza: 3.0646666666666564e-05,  2) Desviación Estandard: 0.005535943159631118


Todas las medidas de tendencia central reflejan un mayor peso de las bebida regular por sobre la dietética.
+ Al analizar ambas series de datos se observa que todas las observaciones de la bebida regular son más pesadas que las observaciones de la dietética.
+ Hay una mayor variación o dispersión de los datos en la categoría regula pero esta no es significativa
________________________________________________________________________________________________________________________________________________________


##### Peniques revisados
A continuación se listan los pesos de peniques acuñados antes y después de 1983.
¿Cuales son las observaciónes en términos de tendencia central?

In [70]:
# Muestra 1
PRE_1983 = [3.1582, 3.0406, 3.0762, 3.0398, 3.1043, 3.1274, 3.0775, 3.1038, 3.1086, 
            3.0586, 3.0603, 3.0502, 3.1028, 3.0522, 3.0546, 3.0185, 3.0712, 3.0717, 
            3.0546, 3.0817, 3.0704, 3.0797, 3.0713, 3.0631, 3.0866, 3.0763, 3.1299, 
            3.0846, 3.0917, 3.0877, 2.9593, 3.0966, 2.9800, 3.0934, 3.1340]

In [71]:
# Muestra 2
POST_1983 = [2.5113, 2.4907, 2.5024, 2.5298, 2.4950, 2.5127, 2.4998, 2.4848, 2.4823, 2.5163, 
             2.5222, 2.5004, 2.5248, 2.5058, 2.4900, 2.5068, 2.5016, 2.4797, 2.5067, 2.5139, 
             2.4762, 2.5004, 2.5170, 2.4925, 2.4876, 2.4933, 2.4806, 2.4907, 2.5017, 2.4950,
             2.4973, 2.5252, 2.4978, 2.5073, 2.4658, 2.4529, 2.5085]

In [72]:
# Medidas de tendencia central
print(f"a. Previous 1983 Central Tendency Measures: 1) media: {mean(PRE_1983)},     2) mediana: {median(PRE_1983)},     3) mode: {mode(PRE_1983)}")
print(f"b. After  1983 Central Tendency Measures: 1) media: {mean(POST_1983)},       2) mediana: {median(POST_1983)},    3) mode: {mode(POST_1983)}")

a. Previous 1983 Central Tendency Measures: 1) media: 3.074782857142857,     2) mediana: 3.0763,     3) mode: 3.0546
b. After  1983 Central Tendency Measures: 1) media: 2.499102702702703,       2) mediana: 2.5004,    3) mode: 2.4907


In [73]:
# Medidas de Variación. Cálculo de Varianza y Desfiación Estandard
print(f"a. Previous 1983:  1) Varianza: {variance(PRE_1983)}, 2) Desviación Estandard: {stdev(PRE_1983)}")
print(f"b. After    1983: 1) Varianza: {variance(POST_1983)},  2) Desviación Estandard: {stdev(POST_1983)}")

a. Previous 1983:  1) Varianza: 0.0015288132268907568, 2) Desviación Estandard: 0.03910004126456591
b. After    1983: 1) Varianza: 0.00027158915915915876,  2) Desviación Estandard: 0.016479962353086818


+ Se observa un mayor peso en las monedas previas a 1983. Esta observación es contundente.
    + La mayoría de las monedas previas son mas pesadas que las post
+ En conjunto de las monedas previas tienen una mayor desviación que las monedas post
    + Se puede haber logrado una mayor precisión en la fabricación de las post
+ Las monedas post 1982 parecen ser mas liianas y con mayor precisión
    + Parace haber una mejora en la fabricación con ahorro de costos


________________________________________________________________________________________________________________________________________________________

#### Palabras de gatos
A veces se utiliza la estadística para comparar o identificar autores de distintos trabajos. 
+ A continuación se incluye una lista con la longitud de 
    + las primeras 20 palabras que escribió Tennessee Williams en el prefacio de The cat on a Hot Tin Roof, 
    + junto con las primeras 20 palabras de The Cat in the Hat, del doctor Seuss.
+ Cat on a Hot Tin Roof: 262214424238422772311
+ The Cat in the Hat: 3333523332422323534 4

+ ¿Parece haber una diferencia en la variación?

In [74]:
CatOnHot = [2,6,2,2,1,4,4,2,4,2,3,8,4,2,2,7,7,2,3,1,1]
CatInHat = [3,3,3,3,5,2,3,3,3,2,4,2,2,3,2,3,5,3,4,4,3]

In [75]:
# Recuento de observaciones y calculo de rangos
print(f"Observaciones:  CatOnHot: {len(CatOnHot)}    CatInHat: {len(CatInHat)}")
print(f"Rangos:         CatOnHot: {max(CatOnHot) - min(CatOnHot)}     CatInHat: {max(CatInHat) - min(CatInHat)}")

Observaciones:  CatOnHot: 21    CatInHat: 21
Rangos:         CatOnHot: 7     CatInHat: 3


In [76]:
# Medidas de tendencia central
print(f"a. CatOnHot: 1) media: {mean(CatOnHot)},     2) mediana: {median(CatOnHot)},     3) mode: {mode(CatOnHot)}")
print(f"b. CatInHat: 1) media: {mean(CatInHat)},     2) mediana: {median(CatInHat)},     3) mode: {mode(CatInHat)}")

a. CatOnHot: 1) media: 3.2857142857142856,     2) mediana: 2,     3) mode: 2
b. CatInHat: 1) media: 3.0952380952380953,     2) mediana: 3,     3) mode: 3


In [77]:
# Medidas de Variación. Cálculo de Varianza y Desfiación Estandard
print(f"a. CatOnHot:  1) Varianza: {variance(CatOnHot)},  2) Desviación Estandard: {stdev(CatOnHot)}")
print(f"b. CatInHat:  1) Varianza: {variance(CatInHat)}, 2) Desviación Estandard: {stdev(CatInHat)}")

a. CatOnHot:  1) Varianza: 4.414285714285715,  2) Desviación Estandard: 2.1010201603710787
b. CatInHat:  1) Varianza: 0.7904761904761904, 2) Desviación Estandard: 0.8890872794479687


+ De la observación de los rangos, las varianzas y las desviaciones estandard, se observa que:
    + hay una mayor variación en las primeras 20 palabras que escribió Tennessee Williams en el prefacio de The cat on a Hot Tin Roof
    + hay una menor variación en las primeras 20 palabras de The Cat in the Hat, del doctor Seuss.

________________________________________________________________________________________________________________________________________________________

#### Exactitud del pronóstico del clima. 
+ En un análisis de la exactitud del pronóstico del clima se comparan las temperaturas máximas reales con las temperaturas máximas pronosticadas un día anterior y con las temperaturas máximas pronosticadas cinco días antes.
    + (máxima real) - (máxima pronosticada un día antes): 2,2,0,0,-3,-2,1,8,1,-2,0,-1,0,1
    + (máxima real) - (máxima pronosticada cinco días antes): 0,-3,2,5,-6,-9,4,-1,6,-2,-2,-1,6,-4
+ ¿La desviación estándar sugiere que las temperaturas pronosticadas un día antes son más exactas que las pronosticadas cinco días antes, como se esperaría?


In [78]:
max1 = [2,2,0,0,-3,-2,1,-2, 8,1,0,-1,0,1]
max5 = [0,-3,2,5,-6,-9,4,-1,6,-2,-2,-1,6,-4]

In [79]:
# Recuento de observaciones y calculo de rangos
print(f"Observaciones:  Pronóstico 1 día antes: {len(max1)}     Pronóstico 5 días antes: {len(max5)}")
print(f"Rangos:         Pronóstico 1 día antes: {max(max1) - min(max1)}     Pronóstico 5 días antes: {max(max5) - min(max5)}")

Observaciones:  Pronóstico 1 día antes: 14     Pronóstico 5 días antes: 14
Rangos:         Pronóstico 1 día antes: 11     Pronóstico 5 días antes: 15


In [80]:
# Medidas de tendencia central
print(f"a. Pronóstico 1 día antes:  1) media: {mean(max1)},     2) mediana: {median(max1)},     3) mode: {mode(max1)}")
print(f"b. Pronóstico 5 días antes: 1) media: {mean(max5)},     2) mediana: {median(max5)},     3) mode: {mode(max5)}")

a. Pronóstico 1 día antes:  1) media: 0.5,     2) mediana: 0.0,     3) mode: 0
b. Pronóstico 5 días antes: 1) media: -0.35714285714285715,     2) mediana: -1.0,     3) mode: -1


In [81]:
# Medidas de Variación. Cálculo de Varianza y Desfiación Estandard
print(f"a. Pronóstico 1 día antes:  1) Varianza: {variance(max1)},  2) Desviación Estandard: {stdev(max1)}")
print(f"b. Pronóstico 5 días antes:  1) Varianza: {variance(max5)}, 2) Desviación Estandard: {stdev(max5)}")

a. Pronóstico 1 día antes:  1) Varianza: 6.884615384615385,  2) Desviación Estandard: 2.6238550616631597
b. Pronóstico 5 días antes:  1) Varianza: 20.554945054945055, 2) Desviación Estandard: 4.533756175065555


+ Si observamos las medias de los errores parecería ser que el error promedio fue menor en el pronostico a 5 días.
+ Pero el error promedio no da una idea absoluta de como fueron los errores de cada estimación 5 días antes.
+ Si boservamos las variaciones de los errores se ve que es mayor la variacíón de 5 días antes.
+ Esta observación es mucho más acertada poque a mayor dispersión de los errores más desasertados fueron los valores pronosticados.

________________________________________________________________________________________________________________________________________________________

#### Regla empírica. 
Las estaturas de un grupo de hombres tienen una distribución normal, con una media de 176 cm y una desviación estándar de 7 cm. Por medio de la regla empírica, ¿cuál es el porcentaje aproximado de hombres entre 
+ a. 169 cm y 183 cm?
+ b. 155 cm y 197 cm?

In [92]:
x=176
s= 7
s2 = 2*s
s3 = 3*s

In [95]:
print(f'El rango de probabilidad del 68% esta entre {x-s} y {x+s}')
print(f'El rango de probabilidad del 95% esta entre {x-s2} y {x+s2}')
print(f'El rango de probabilidad del 99,7% esta entre {x-s3} y {x+s3}')

El rango de probabilidad del 68% esta entre 169 y 183
El rango de probabilidad del 95% esta entre 162 y 190
El rango de probabilidad del 99,7% esta entre 155 y 197


La regla empirica de la distribución normal sostiene los valores de 68%, 95%, y 99,7% para 1S, 2S, y 3S de la distribución normal.
+ En este caso la media x=176cm y la desviación estandar= 7. Entonces para:
    + 1s el rango de 68% de probabilidad es 169<=176<=183
    + 2s el rango de 95% de probabilidad es 162<=176<=190
    + 3s el rango de 99% es 155<=176<=197
+ Lo anterior, quiere decir entonces que:
+ a. El ´porcentaje aproximado para hombres de entre 169cm y 183cm es 68%.
+ b. El ´porcentaje aproximado para hombres de entre 155cm y 197cm es 99%.

________________________________________________________________________________________________________________________________________________________

#### Coeficiente de variación. 
Utilice los siguientes datos muestrales para calcular elcoeficiente de variación de cada muestra; después, compare los resultados.
+ Estaturas (en pulgadas) de hombres: 71 66 72 69 68 69
+ Largo (en mm) de huevos de ave: 19.7 21.7 21.9 22.1 22.1 22.3 22.7 22.9 23.9

In [86]:
# Calculo de media y desviación estandard de la primera muestra
mean([71,66,72,69,68,69]),stdev([71,66,72,69,68,69])

(69.16666666666667, 2.136976056643281)

In [87]:
# Calculo de media y desviación estandard de la segunda muestra
mean([ 19.7, 21.7, 21.9, 22.1, 22.1, 22.3, 22.7, 22.9, 23.9]), stdev([19.7, 21.7, 21.9, 22.1, 22.1, 22.3, 22.7, 22.9, 23.9])

(22.144444444444446, 1.130388330520878)

In [88]:
# Calculo los coeficientes de variación de ambas muestras y los comparo
coefestaturas= stdev([71,66,72,69,68,69])/mean([71,66,72,69,68,69])*100 
coeflargo=stdev([19.7, 21.7, 21.9, 22.1, 22.1, 22.3, 22.7, 22.9, 23.9])/mean([ 19.7, 21.7, 21.9, 22.1, 22.1, 22.3, 22.7, 22.9, 23.9])*100
coefestaturas, coeflargo

(3.089603937315587, 5.104613635066684)

In [85]:
print(f'Coeficientes de variación: Estaturas = {coefestaturas}%. Largos= {coeflargo}%')

Coeficientes de variación: Estaturas = 3.089603937315587%. Largos= 5.104613635066684%


El coeficiente de variación de estaturas es menor que el coeficiente d variación de largos.
Esto significa que las observaciones de las estaturas tienen una variación menor que las observaciones de la muestra de largos.